### DISCUSSION TITLE
upstash session implementation

### DISCUSSION THREAD

**Discussion by omorShahriar** [mitchelvanbever/remix-auth-supabase#19] [Q&A]
session file made by supabase is pretty big to store in cookie. It has worsen the performance in my project. There's really long latency in the app now. Can anyone suggest me some alternative for session storage? Like how to implement upstash redis in this scenario.

**Comment by rphlmr** [comment]
Hello,
I don't know anything about Redis but you can create your own session storage and  connect it to database (and I guess, Redis)

Check this, I hope it help
https://remix.run/docs/en/v1/api/remix#createsessionstorage

PS : thanks for your feed-back. I don't notice issue on my project but it's good to know.

**Reply by rphlmr** [reply]
Maybe if  https://github.com/mitchelvanbever/remix-auth-supabase/pull/20 is merged, you'll see a perf bonus 😇

**Reply by omorShahriar** [reply]
thanks for responding to my query. The latency issue is fixed now. It was happening since my server location and supabase server's location was really far away
